adoc/chapters/device_compiler.adoc

```diff
@@ -387,6 +281,25 @@ literal with greater value.
 In addition, for each <<backend>> supported, the preprocessor macros described
 in <<sec:backends>> must be defined by all conformant implementations.
 
+[[sec:standard-library-support]]
+== {cpp} standard library support
+
```

Comment:
This section is the first time we specifically guarantee that anything in the `std` namespace can be used in device code.  I think we need to prefix this section with an introductory statement something like:

> In general, any use of functions or types from the `std` namespace in device code produces undefined behavior.  However, the functions and types listed in this section are guaranteed to have defined semantics when used in device code.

Reply:
Added your proposed paragraph in da1c0db2e23e1ad79f35c6970975ec49fd370201.